### PR TITLE
feat: add full text search to docs

### LINKED ISSUE #11597: Unable to search docs with query: signup
Needs more info on how signup works.

Especially  `options` field, where is it stored on DB. How to access that data?

```
const { data, error } = await supabaseClient.auth.signUp({
				email: email,
				password: password,
				options: {
					data: {
						name: name,
						surname: surname
					}
				}
			});
```

### LINKED ISSUE #12149: Unable to search docs with query: 54324 or query:inbucket
My search may be failing, or the information may be missing. When we are developing locally via the > supabase start, nothing tells me how to use authentication. I want to follow the TDD ( test driven development ) approach to code. This would be very helpful.

Reading the docs, I found out there is an email testing server called InBucket. Using http://localhost:54324/ brought me to that UI. Progress, but the developer journey should have included that. You may be losing curious users on this who hit a roadblock.

### LINKED ISSUE #19161: Docs search not well indexed?
Go to the docs at supabase.com/docs

click on the search box

enter 'local'

click search...

result = no docs found

### PR DESCRIPTION
**Pull request by charislam**
## What kind of change does this PR introduce?

**Feature**

Adds a full-text search endpoint for docs search, to provide more comprehensive and faster results. Each docs search now performs both FTS and embeddings-based search, and displays results as soon as the first set of results is returned.

**Refactor**

Slight refactor of the embeddings search to make only one trip to the database, and transfer less content in the JSON. This now uses a different function than the RAG step for Supabase AI.

**Bug fix**

Fixes a bug where some search results were being filtered out by the `cmdk` menu.

Removes an obsolete guide. Content is now elsewhere and it messes up search results.

## What is the new behavior?

https://github.com/supabase/supabase/assets/26616127/569c04c6-30d8-4644-80b2-c866d5695ebf

## Fixes

Resolves #11597
Resolves #19161 
Resolves #12149

### CODE REVIEW

File: apps/docs/pages/guides/database.mdx
Comment (charislam) [inline apps/docs/pages/guides/database.mdx:1]: This file is obsolete and a duplicate of an existing page, which messes up search results

File: supabase/functions/common/database-types.ts [hunk 1]
```diff
@@ -84,19 +100,74 @@ export interface Database {
...
+        }
+        Insert: {
+          details?: string | null
+          id?: number
+        }
+        Update: {
+          details?: string | null
+          id?: number
+        }
+        Relationships: []
```
Comment (charislam) [inline supabase/functions/common/database-types.ts:133]:
```suggestion
```

**Comment by MildTomato** [comment]
@kamilogorek hey 👋 we've got github integration breaking.

We've deleted and recreated the Supabase Preview Branch for this git branch but it seems like the new Preview Branch doesn't know about this PR

**Comment by MildTomato** [comment]
closing PR and reopening for branch testing

**Review by saltcod — APPROVED** [review]

**Pull request merged**